src/JustSaying/AwsTools/QueueCreation/SqsReadConfiguration.cs

```diff
@@ -14,48 +20,61 @@ public SqsReadConfiguration(SubscriptionType subscriptionType)
             ErrorQueueRetentionPeriod = JustSayingConstants.MaximumRetentionPeriod;
             VisibilityTimeout = JustSayingConstants.DefaultVisibilityTimeout;
             RetryCountBeforeSendingToErrorQueue = JustSayingConstants.DefaultHandlerRetryCount;
+            SubscriptionConfig = new SubscriptionConfigBuilder();
         }
 
+        public SubscriptionConfigBuilder SubscriptionConfig { get; private set; }
         public SubscriptionType SubscriptionType { get; private set; }
 
-        public string QueueName { get; set; }
         public string TopicName { get; set; }
         public string PublishEndpoint { get; set; }
 
-        public int? MaxAllowedMessagesInFlight { get; set; }
-        public IMessageProcessingStrategy MessageProcessingStrategy { get; set; }
-        public Action<Exception, Amazon.SQS.Model.Message> OnError { get; set; }
         public string TopicSourceAccount { get; set; }
         public IMessageBackoffStrategy MessageBackoffStrategy { get; set; }
         public string FilterPolicy { get; set; }
+        public string SubscriptionGroupName { get; set; }
 
-        public override void Validate()
+        protected override void OnValidate()
         {
-            ValidateSqsConfiguration();
-            ValidateSnsConfiguration();
-        }
+            if (SubscriptionType == SubscriptionType.ToTopic)
+            {
+                if (string.IsNullOrWhiteSpace(TopicName))
+                {
+                    throw new ConfigurationErrorsException("Invalid configuration. Topic name must be provided.");
```

Comment:
Yeah it's a good point. I think where that kind of validation is most useful is when you want to show all of the errors to the user at the same time, e.g. from an API to a client. In this case, we're failing fast when there is a configuration issue so perhaps building an error list isn't as useful as failing as soon as the developer has made a mistake?

This is also the way the configuration validation already worked, so we didn't change it. It's worth considering that JustSayingStack makes use of some of these types at a pretty low level, so the more we change them, the harder it will be to update it, and also those that depend on it 🙂.

Reply:
> isn't as useful as failing as soon as the developer has made a mistake?
But they make a mistake (possibly many) and then run their unit test (or worse: wait for CI and CD to deploy to some real environment). I think we'd rather tell them about all their mistakes at once, so that they can stay "in the zone".

Reply:
Had a discussion with @gkinsman, @martincostello, @stuart-lang and we felt this would be too much effort for too little gain. Existing pattern is good enough.